website/docs/r/vm_affinity_rule.html.markdown

```diff
@@ -0,0 +1,125 @@
+---
+layout: "vcd"
+page_title: "vCloudDirector: vcd_vm_affinity_rule"
+sidebar_current: "docs-vcd-resource-vm-affinity-rule"
+description: |-
+  Provides a vCloud Director VM affinity rule resource. This can be
+  used to create, modify, and delete VM affinity and anti-affinity rules.
+---
+
+# vcd\_vm\_affinity\_rule
+
+Provides a vCloud Director VM affinity rule resource. This can be
+used to create, modify, and delete VM affinity and anti-affinity rules.
+
+Supported in provider *v2.9+*
+
+~> **Note:** The vCD UI defines two different entities (*Affinity Rules* and *Anti-Affinity Rules*). This resource combines both
+entities: they are differentiated by the `polarity` property (see below).
+
+## Example Usage
+
+```hcl
+data "vcd_vapp" "Test_EmptyVmVapp1" {
+  name = "Test_EmptyVmVapp1"
+}
+
+data "vcd_vapp_vm" "Test_EmptyVm1a" {
+  vapp_name = data.vcd_vapp.Test_EmptyVmVapp1.name
+  name      = "Test_EmptyVm1a"
+}
+
+data "vcd_vapp_vm" "Test_EmptyVm1b" {
+  vapp_name = data.vcd_vapp.Test_EmptyVmVapp1.name
+  name      = "Test_EmptyVm1b"
+}
+
+data "vcd_vapp_vm" "Test_EmptyVm1c" {
+  vapp_name = data.vcd_vapp.Test_EmptyVmVapp1.name
+  name      = "Test_EmptyVm1c"
+}
+
+resource "vcd_vm_affinity_rule" "Test_VmAffinityRule1" {
+  name     = "Test_VmAffinityRule1"
+  required = true
+  enabled  = true
+  polarity = "Affinity"
+
+  virtual_machine_ids = [
+    data.vcd_vapp_vm.Test_EmptyVm1a.id,
+    data.vcd_vapp_vm.Test_EmptyVm1b.id,
+    data.vcd_vapp_vm.Test_EmptyVm1c.id
+  ]
+}
+```
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional) The name of organization to use, optional if defined at provider level. Useful when connected as sysadmin working across different organisations
+* `vdc` - (Optional) The name of VDC to use, optional if defined at provider level
+* `name` - (Required) The name of VM affinity rule. Duplicates are allowed, although the name can be used to retrieve
```

Comment:
Retrieve? Please rephrase/add a remark what retrieve means in this context (importing?).

Reply:
We are talking about the resource here, but eventually that resource becomes a data source, and we need to `retrieve` it. We have a similar predicament when we import.
The caveat must be here, because the server doesn't enforce uniqueness, and thus users must take care of the names if they want to use them later to find the entity.

Reply:
Added a note about the above